### PR TITLE
[Guide] Search Added

### LINKED ISSUE #353: [Guide] Search
Create `guides/search.md` similar to the [Ruby guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/guides/search.md)

You must assure that: 
- All code-blocks, when run from start to finish in order they appear in the guide, do not throw any errors. If you want to demonstrate that certain statements will throw errors, they must be wrapped in a `try/catch` and print out the expected error.
- All code-blocks are properly marked with the language it's written in (i.e.  ` ```ruby `)
- The instructions, though mostly identical to the Ruby guide, are suitable for this repo.

### PR DESCRIPTION
**Pull request by jdrysdaleandesrson**
### Description
Added [Guide] Search. Heavily based off of the ruby guide search.

### Issues Resolved
#353 

No search documentation for opensearch-py version

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Jonathan Anderson <jonadrys@gmail.com>

### CODE REVIEW

File: guides/search.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+# Create an OpenSearch client with appropriate hosts and connection details
+client = OpenSearch(hosts=['localhost'])
+# Define the search query with scroll and pagination options
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "scroll": "1m",  # Set the scroll duration to 1 minute
```
Comment (VachaShah) [inline guides/search.md:205]: `scroll` is a parameter for the search request. It should be removed from here and added in L209 as follows:

```
page_1 = client.search(
    index='movies',
    scroll='1m',
    body=search_body
)
```

File: guides/search.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+page_3 = client.search(
+    size=2,
+    body=pit_search_body.copy().update({'search_after': page_2[-1]['sort']})
+)['hits']['hits']
+# Print out the titles of the first 3 pages of results
+print([hit['_source']['title'] for hit in page_1])
+print([hit['_source']['title'] for hit in page_2])
+print([hit['_source']['title'] for hit in page_3])
+# Delete the point in time
+client.delete_pit(body={'pit_id': pit['pit_id']})
```
Comment (VachaShah) [inline guides/search.md:282]: The API is `delete_point_in_time`. See https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#point-in-time-api

File: guides/search.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    "sort": [
+        {
+            "year": {
+                "order": "asc"
+            }
+        }
+    ]
+}
+# Create a point in time
+pit = client.create_pit(index='movies', keep_alive='1m')
```
Comment (VachaShah) [inline guides/search.md:259]: The API is `create_point_in_time`. See https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#point-in-time-api.

**Review by VachaShah — COMMENTED** [review]
Thank you for adding this! I tried out the code and had a few comments.

File: guides/search.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+# Get the first 3 pages of results
+page_1 = client.search(size=2, body=pit_search_body)['hits']['hits']
+page_2 = client.search(
+    size=2,
+    body=pit_search_body.copy().update({'search_after': page_1[-1]['sort']})
+)['hits']['hits']
+
+page_3 = client.search(
+    size=2,
+    body=pit_search_body.copy().update({'search_after': page_2[-1]['sort']})
```
Comment (VachaShah) [inline guides/search.md:275]: At this line, I get the following error:

```
Traceback (most recent call last):
  File "opensearch-py-search.py", line 215, in <module>
    body=pit_search_body.copy().update({'search_after': page_2[-1]['sort']})
KeyError: 'sort'
```

**Review by VachaShah — COMMENTED** [review]
This PR also needs a Changelog entry.

**Comment by VachaShah** [comment]
@jdrysdaleandesrson Can you fix your DCO for the last commit? You can do it using:

```
git commit --amend -s
```

You can also group your commit using `git rebase`.

Also, please add a Changelog entry for this PR in [CHANGELOG](https://github.com/opensearch-project/opensearch-py/blob/main/CHANGELOG.md).

**Comment by jdrysdaleandesrson** [comment]
@VachaShah thanks for the feedback. I believe the issues you discussed have been changed. lmk if there is anything else.

**Comment by VachaShah** [comment]
> @VachaShah thanks for the feedback. I believe the issues you discussed have been changed. lmk if there is anything else.

This looks great. Can you add the Changelog entry as well?

**Comment by jdrysdaleandesrson** [comment]
@VachaShah just updated the Changelog

**Comment by VachaShah** [comment]
> @VachaShah just updated the Changelog

@jdrysdaleandesrson It should be part of this PR. There is changelog verifier workflow on each PR which runs to verify changelog which will fail if changelog is not added.

**Comment by jdrysdaleandesrson** [comment]
@VachaShah sorry about that havent worked with ChangeLogs before. Thanks for your patience should be fixed and updated now.

**Review by VachaShah — APPROVED** [review]
Thank you @jdrysdaleandesrson for your patience and working through the comments!

**Pull request merged**